catroid/src/main/java/org/catrobat/catroid/utils/IconsUtil.java

```diff
@@ -0,0 +1,386 @@
+					drawable = context.getResources().getDrawable(R.drawable.phiro_neg);
+				} else if (category.equals(context.getString(R.string.category_raspi))) {
+					drawable = context.getResources().getDrawable(R.drawable.raspberrypi_neg);
+				} else if (category.equals(context.getString(R.string.category_event))) {
+					drawable = context.getResources().getDrawable(R.drawable.event_neg);
+				} else if (category.equals(context.getString(R.string.category_pen))) {
+					drawable = context.getResources().getDrawable(R.drawable.pen_neg);
+				}
+			} else {
+				if (category.equals(context.getString(R.string.category_control))) {
+					drawable = context.getResources().getDrawable(R.drawable.control_pos);
+				} else if (category.equals(context.getString(R.string.category_motion))) {
+					drawable = context.getResources().getDrawable(R.drawable.motion_pos);
+				} else if (category.equals(context.getString(R.string.category_sound))) {
+					drawable = context.getResources().getDrawable(R.drawable.sound_pos);
+				} else if (category.equals(context.getString(R.string.category_looks))) {
+					drawable = context.getResources().getDrawable(R.drawable.look_pos);
+				} else if (category.equals(context.getString(R.string.category_user_bricks))) {
+					drawable = context.getResources().getDrawable(R.drawable.userbrick_pos);
+				} else if (category.equals(context.getString(R.string.category_data))) {
+					drawable = context.getResources().getDrawable(R.drawable.data_pos);
+				} else if (category.equals(context.getString(R.string.category_lego_nxt))) {
+					drawable = context.getResources().getDrawable(R.drawable.legonxt_pos);
+				} else if (category.equals(context.getString(R.string.category_arduino))) {
+					drawable = context.getResources().getDrawable(R.drawable.arduino_pos);
+				} else if (category.equals(context.getString(R.string.category_drone))) {
+					drawable = context.getResources().getDrawable(R.drawable.ardrone_pos);
+				} else if (category.equals(context.getString(R.string.category_phiro))) {
+					drawable = context.getResources().getDrawable(R.drawable.phiro_pos);
+				} else if (category.equals(context.getString(R.string.category_raspi))) {
+					drawable = context.getResources().getDrawable(R.drawable.raspberrypi_pos);
+				} else if (category.equals(context.getString(R.string.category_event))) {
+					drawable = context.getResources().getDrawable(R.drawable.event_pos);
+				} else if (category.equals(context.getString(R.string.category_pen))) {
+					drawable = context.getResources().getDrawable(R.drawable.pen_pos);
+				}
+			}
+
+			if (drawable != null) {
+				if (isLargeSize()) {
+					drawable.setBounds(largeIconSizeCategory);
+				} else {
+					drawable.setBounds(smallIconSizeCategory);
+				}
+				textView.setCompoundDrawables(drawable, null, null, null);
+			}
+		}
+	}
+
+	public static void addIcons(ViewGroup viewGroup, String category) {
+		if (isActivated()) {
```

Comment:
I would replace it with
if (!isActivated()) {
  return; 
} ..
as there are many nested conditions/loops to keep the a code a bit clearer